File: 240128/최고의 13위치/best-place-of-13.py
N=int(input())

graph=[list(map(int,input().split())) for _ in range(N)]

ans=0

for i in range(N):
    for j in range(N-2):
        ans=max(ans,graph[i][j]+graph[i][j+1]+graph[i][j+2])

print(ans)

File: 260105/최고의 13위치/best-place-of-13.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');

const n = Number(input[0]);
const arr = input.slice(1).map(line => line.trim().split(' ').map(Number));

// Please Write your code here.

let ans = 0;

for(let i=0; i<n; i++){
    for(let j=0; j<n-2; j++){
        let cnt = 0;
        cnt+=arr[i][j]+arr[i][j+1]+arr[i][j+2]
        ans=Math.max(ans,cnt);
    }
}

console.log(ans)